storage/time-series/influxdb/storer/app/App.js

```diff
@@ -0,0 +1,247 @@
+    */
+  constructor() {
+    logger.debug('constructor: instantiate app...');
+    try {
+      this.kafkaConsumer = new KafkaConsumer();
+      this.influxOrgs = new InfluxOrgs(
+        configInflux.url,
+        configInflux['default.token'],
+        configInflux['default.user'],
+        configInflux['default.password'],
+        configInflux['default.organization'],
+        configInflux['default.bucket'],
+        configInflux['retention.hrs'],
+      );
+      this.influxWriter = new InfluxWriter(
+        configInflux.url,
+        configInflux['default.token'],
+        configInflux['default.bucket'],
+        configInfluxWriteOptionsCamelCase,
+      );
+      this.influxMeasurement = new InfluxMeasurement(
+        configInflux.url,
+        configInflux['default.token'],
+        configInflux['default.bucket'],
+      );
+      this.influxState = new InfluxState(
+        configInflux.url,
+      );
+    } catch (e) {
+      logger.error('constructor:', e);
+      throw e;
+    }
+  }
+
+  /**
+   * Initialize the kafka and influxdb (including HeathChecker and Shutdown)
+   */
+  async init() {
+    logger.info('init: Initializing the influxdb-storer ...');
+    try {
+      this.createHealthCheckers();
+      this.defineShutdown();
+
+      // initializes kafka consumer
+      await this.kafkaConsumer.init();
+
+      const kafkaIsConnected = await this.kafkaConsumer.isConnected();
+      if (!kafkaIsConnected) {
+        throw new Error('Kafka is not ready');
+      }
+      const influxIsReady = await this.influxState.isReady();
+      if (!influxIsReady) {
+        throw new Error('Influxdb is not ready');
+      }
+
+      // create the initial org, user and bucket
+      await this.influxOrgs.initOnboarding();
+
+      // associate by callback kafka consumers and influxdb actions
+      this.initCallbacksTenantToCreateAndDelOrgs();
+      this.initCallbacksDevicesCreateAndDelData();
+
+      // Signals to the health check service that the service is ready.
+      logger.debug('Init: Signaling that the service is ready...');
+      ServiceState.signalReady();
+      logger.debug('Init: ...signaled.');
+    } catch (e) {
+      logger.error('init:', e);
+      await ServiceState.shutdown();
+    }
+    logger.info('init: ... service initialized.');
+  }
+
+  /**
+   * Associates callbacks with kafka devices events to manipulation of influxdb data
+   */
+  initCallbacksDevicesCreateAndDelData() {
+    // create callback to handle receive data
+    const boundInfluxWriter = this.influxWriter.writer.bind(this.influxWriter);
+    const callbackWriteData = async (tenant, deviceid, timestamp, attrs) => {
+      logger.debug('callbackWriteData: init');
+      logger.debug(`callbackWriteData: tenant=${tenant}`);
+      logger.debug(`callbackWriteData: deviceid=${deviceid}`);
+      logger.debug(`callbackWriteData: timestamp=${timestamp}`);
+      try {
+        await boundInfluxWriter(tenant, deviceid, attrs, timestamp);
```

Comment:
Do you think I should keep a list of created devices (not ignoring the device creation event anymore) and just try to write if that device exists? I think not....
Do you think I should keep a list of created tenants and only try to write if this org exists? I think not....